mmcv/runner/iter_based_runner.py

```diff
@@ -34,6 +34,7 @@ def __next__(self):
             self._epoch += 1
             if hasattr(self._dataloader.sampler, 'set_epoch'):
                 self._dataloader.sampler.set_epoch(self._epoch)
+            time.sleep(2)  # Prevent possible deadlock during epoch transition
```

Comment:
In my experience, I never met this problem, and 2 seconds might be unfriendly to small datasets.

Reply:
We met this problem in MME.

Reply:
OK. It doesn't have much effect on me. Might hear others' opinions.

Reply:
In fact, I have NOT met any deadlock caused by the dataloader. If this modification can solve your problem in training, we can merge it.

Reply:
Hi @RangiLyu, could you explain what is MME? Can I reproduce deadlock?